game/src/core/Game.java

```diff
@@ -262,6 +263,27 @@ public static Optional<System> add(final System system) {
     return ECSManagment.add(system);
   }
 
+  /**
+   * Add {@link System}s to the game.
+   *
+   * <p>If a System is added to the game, the {@link System#execute} method will be called every
+   * frame.
+   *
+   * <p>Additionally, the systems will be informed about all new, changed, and removed entities.
+   *
+   * <p>The game can only store one system of each system type.
+   *
+   * @param systems the Systems to add
+   * @return an optional set that contains the previous existing systems of the given systems
+   *     classes, if exists
+   * @see Set
+   * @see Optional
+   * @see System
+   */
+  public static Set<Optional<System>> add(final Set<System> systems) {
```

Comment:
@tgrothe Bitte schau Dir doch mal die anderen Methoden in Game an. Machen die wirklich etwas selbst? Das vorhandene Paradigma scheint doch zu sein, alles an externe Hilfsklassen weiterzuleiten und die die Arbeit machen zu lassen. (Nicht dass ich das unbedingt gut finde, aber es sollte konsistent bleiben.)

Reply:
> schau doch mal eine Zeile weiter runter, diese Methode delegiert quasi an `Game.add` bzw. an `ECSManagment.add(system)`, und gibt dessen Ergebnis zurück ... mehr soll semantisch und prinzipiell doch auch nicht geschehen?

Wenn ich das richtig in Erinnerung habe, dann hast Du da sowas wie `systems.stream().map(Game::add).collect(Collectors.toSet());` geschrieben. Das ist in meinen Augen deutlich mehr als nur Delegation. 

(Ja, Du rufst darin irgendwo auch `Game.add()` auf. Aber Du hast hier eine Funktionalität implementiert, die es bei den anderen `Game.add()`-Methoden nicht gibt - die rufen einfach mit ihren Parametern die entsprechende Methode aus `ECSManagment` auf, und _dort_ passiert die eigentliche Verarbeitung.)

Reply:
> (Ja, Du rufst darin irgendwo auch `Game.add()` auf. Aber Du hast hier eine Funktionalität implementiert, die es bei den anderen `Game.add()`-Methoden nicht gibt - die rufen einfach mit ihren Parametern die entsprechende Methode aus `ECSManagment` auf, und _dort_ passiert die eigentliche Verarbeitung.)

Das stimmt, die eigentliche Verarbeitung geschieht in `ECSManagment`, aber die andere Add-Methode macht genau dies auch ...

Moment, ich suche ...

https://github.com/Dungeon-CampusMinden/Dungeon/blob/877eeaf8e86736afc1fa4824b16f4f42eeec624c/game/src/core/Game.java#L261-L263

Also, es wird an `ECSManagment` weitergegeben und dann das Ergebnis zurückgegeben.

Nur haben wir bei mehreren, hinzuzufügenden System natürlich auch mehrere Ergebnisse ...

Aber ich habe auch noch nicht ganz verstanden, weshalb man generell hierbei einen Rückgabewert braucht.

Reply:
Ja eben, das ist doch der Punkt?! Die Verarbeitung passiert in ECSManagement, nicht in Game. Und das möchte ich so beibehalten.

Die Rückgabe sind lt. Doku die ggf. bisher gesetzten Systems, die jetzt ersetzt werden. Ob man die braucht, weiss ich nicht. Aber das ist ja hier auch gar nicht der Punkt.

Reply:
Ach so, jetzt habe ich es verstanden. ;) Ich rufe die falsche/verkehrte Add-Methode in dem Stream-Konstrukt auf ...

Reply:
> Ach so, jetzt habe ich es verstanden. ;) Ich rufe die falsche/verkehrte Add-Methode in dem Stream-Konstrukt auf ...

????? du hantierst in game aktiv mit den parametern, anstatt die nach ecsmanagement direkt durchzureichen und **dort** das verarbeiten zu machen. das passt nicht zu den anderen game-methoden und stört. es ist aber nicht, dass du eine verkehrte methode oder so aufrufst.

Reply:
> > Ach so, jetzt habe ich es verstanden. ;) Ich rufe die falsche/verkehrte Add-Methode in dem Stream-Konstrukt auf ...
> 
> ????? du hantierst in game aktiv mit den parametern, anstatt die nach ecsmanagement direkt durchzureichen und **dort** das verarbeiten zu machen. das passt nicht zu den anderen game-methoden und stört. es ist aber nicht, dass du eine verkehrte methode oder so aufrufst.

Du meinst, die komplette Methode `Set<Optional<System>> add(final Set<System> systems)` sollte "weg" bzw. in ecsmanagement sein?

Reply:
Und sorry, es ist schon spät, ich kann morgen noch mal hiernach schauen.